fvm/env.go

```diff
@@ -331,6 +337,10 @@ func (e *hostEnv) EmitEvent(event cadence.Event) error {
 		Payload:          payload,
 	}
 
+	if fvmEvent.IsServiceEvent(event, e.ctx.Chain) {
+		e.serviceEvents = append(e.serviceEvents, flowEvent)
+	}
+
```

Comment:
do we want to add it to events as well or we should return here?

Reply:
I think all events should be part of events emitted. We just intercept some special ones.